digdag-server/src/main/java/io/digdag/server/rs/ProjectResource.java

```diff
@@ -280,18 +287,26 @@ public RestWorkflowDefinition getWorkflowByName(@PathParam("id") int projId, @Pa
     @Path("/api/projects/{id}/archive")
     @Produces("application/gzip")
     public byte[] getArchive(@PathParam("id") int projId, @QueryParam("revision") String revName)
-        throws ResourceNotFoundException
+        throws ResourceNotFoundException, StorageFileNotFoundException
     {
         ProjectStore ps = rm.getProjectStore(getSiteId());
-        StoredProject proj = ensureNotDeletedProject(ps.getProjectById(projId));
-        StoredRevision rev;
-        if (revName == null) {
-            rev = ps.getLatestRevision(proj.getId());
+        ensureNotDeletedProject(ps.getProjectById(projId));
+
+        Optional<InputStream> in = archiveManager.openArchive(ps, projId, revName);
+        if (!in.isPresent()) {
+            // ArchiveType of this revision is NONE
+            return new byte[0];
         }
         else {
-            rev = ps.getRevisionByName(proj.getId(), revName);
+            // TODO how to return InputStream with Content-Length header so that server doesn't have to read entire contents in memory?
```

Comment:
`Content-Length` is not strictly required so we could just return the `InputStream` directly or use `StreamingOutput` etc.

And if desired we could add a method to `ArchiveManager` to query for the size of the archive so that the `Content-Length` can be set.